### PR TITLE
chore: Add initial content to the process chapter

### PR DESCRIPTION
**Pull request by bebraw**

### CODE REVIEW

File: manuscript/infrastructure/01-processes.md [hunk 1]
```diff
@@ -4,22 +4,50 @@ Each project has processes around it. Services like [GitHub](https://github.com/
...
+In a small project you might be aware of everything you need to do. You implement what you intend and publish. That's it.
 
-TODO: Issue and PR templates, https://github.com/serverless/serverless/blob/master/.github/ISSUE_TEMPLATE.md
+It gets more complex if you plan for the future or if the users of the package discover problems. This where **issue tracking** is required.
 
-## Designing Process and Labels
+### What Issue Data to Capture
 
-TODO
+Depending on the system, an issue contains at least the following data: title, description, labels or tags. It may also be assigned to a milestone or to specific project members, or projects in GitHub.
```
Comment (sapegin) [inline manuscript/infrastructure/01-processes.md]: Are they projects, not teams or something?

File: manuscript/infrastructure/01-processes.md [hunk 2]
```diff
@@ -4,22 +4,50 @@ Each project has processes around it. Services like [GitHub](https://github.com/
...
+
+The point is to capture this data so that development decisions can be made. An issue itself can be a feature request, a bug report, or sometimes a support request. It's up to project policies on how to deal with support in particular and in some cases support requests are pushed to specific channels while the official issue tracker of the project is limited only for development purposes.
```
Comment (sapegin) [inline manuscript/infrastructure/01-processes.md]: “pushed to specific channels” is too abstract, better to give an example or just skip, you’re discussing support in details later.

File: manuscript/infrastructure/01-processes.md [hunk 3]
```diff
@@ -4,22 +4,50 @@ Each project has processes around it. Services like [GitHub](https://github.com/
...
+
+### Capturing the Right Data
+
+Especially popular projects are constrained by the amount of maintainers they have. This means that the demand for support in different means can exceed the bandwidth of the team. Managing issue tracking well is one of the keys to dealing with this problem.
+
+Poorly formatted issues waste time. In the worst case you have to dig the information you require to debug the problem. The problem can be solved by improving the standard of issues and not accepting poor ones.
```
Comment (sapegin) [inline manuscript/infrastructure/01-processes.md]: I’d say incomplete issues are a bigger problem.

File: manuscript/infrastructure/01-processes.md [hunk 4]
```diff
@@ -4,22 +4,50 @@ Each project has processes around it. Services like [GitHub](https://github.com/
...
+
+In GitHub this can be achieved by using an [issue template](https://github.com/serverless/serverless/blob/master/.github/ISSUE_TEMPLATE.md) that's included within a repository. When a user fills in an issue, the provided template will work as a starting point. Unfortunately, given it's in text format, it doesn't solve the problem entirely. Bots discussed in the *Automation* chapter solve this problem as they can close poor quality issues.
```
Comment (sapegin) [inline manuscript/infrastructure/01-processes.md]: I’d say bots are trying to solve the issue or helping ;-)

File: manuscript/infrastructure/01-processes.md [hunk 5]
```diff
@@ -4,22 +4,50 @@ Each project has processes around it. Services like [GitHub](https://github.com/
...
+
+## Pull Requests
+
+Instead of requesting for features or reporting bugs, sometimes the users may scratch their own itch and provide **pull requests** to a project. The point is to provide a small patch to the project that can then be merged provided it's high enough quality.
```
Comment (sapegin) [inline manuscript/infrastructure/01-processes.md]: provide provided

File: manuscript/infrastructure/01-processes.md [hunk 6]
```diff
@@ -4,22 +4,50 @@ Each project has processes around it. Services like [GitHub](https://github.com/
...
+
+## Designing Development Process
+
+An important part of development practices is designing a development process. Especially early on this can be ad hoc as you don't have to worry about team productivity. Once more people become involved, you have to make sure they can contribute in a meaningful manner.
+
+This is where managing issue tracking and pull requests well becomes vital. You should design a system for issues where you set them in a particular way. For example, each issue could gain a priority (0-4 for example), categorization (bug, feature request), and difficulty (0-3 for example). The labels you need depend on the type of the project and there's no one right answer on which labels to use.
+
+In GitHub pull requests may use labels as well. Most importantly it's a chance for you to apply reviews on code. That is your chance to discuss the implementation until it's in a satisfactory state. It's also a chance to grow new contributors to your project. They may even become maintainers one day and assume your position in the project.
```
Comment (sapegin) [inline manuscript/infrastructure/01-processes.md]: The first sentence feels out of place here.

File: manuscript/infrastructure/01-processes.md [hunk 7]
```diff
@@ -4,22 +4,50 @@ Each project has processes around it. Services like [GitHub](https://github.com/
...
 
 ## Designing a Support Model
 
-TODO: GitHub Issues vs. Stack Overflow.
+Once a project reaches certain scale, managing support becomes a problem. GitHub's issue tracker works up to a point. Sometimes bigger projects push questions to services like Stack Overflow. This is also where commercial support may be provided as there is no obligation to help free even if the code is freely available.
```
Comment (sapegin) [inline manuscript/infrastructure/01-processes.md]: ”Push” feels like some kind of automation, but it’s usually a manual process if I haven’t missed some great tool ;-)

File: manuscript/infrastructure/01-processes.md (hunk 1 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Projects as in the new project tab. I think I will simplify this bit somehow.

File: manuscript/infrastructure/01-processes.md (hunk 3 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Yeah, good point. I'll add a mention.

File: manuscript/infrastructure/01-processes.md (hunk 2 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Yup, agreed.

File: manuscript/infrastructure/01-processes.md (hunk 4 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Yup, better soften the statement.

File: manuscript/infrastructure/01-processes.md (hunk 5 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Yup, replaced latter.

File: manuscript/infrastructure/01-processes.md (hunk 1 above)
Comment (sapegin) [reply manuscript/infrastructure/01-processes.md]: I thought you mean this thing 

image

File: manuscript/infrastructure/01-processes.md (hunk 6 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Yeah, better drop.

File: manuscript/infrastructure/01-processes.md (hunk 7 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Yup, I'll use a different verb.

File: manuscript/infrastructure/01-processes.md (hunk 1 above)
Comment (bebraw) [reply manuscript/infrastructure/01-processes.md]: Ah, maybe teams should be discussed separately. Good point.

**Pull request merged**